File: config/function_crud.go
package config

import (
	"database/sql"
	"fmt"
	"log"
)

func Insert(db *sql.DB, table string, data map[string]string) (status string) {
	fields := ""
	values := ""
	for f, v := range data {
		fields += fmt.Sprintf(", %s", f)
		values += fmt.Sprintf(", '%s'", v)
	}
	query := fmt.Sprintf("INSERT INTO %s (%s) VALUES(%s)", table, fields[1:], values[1:])
	_, err := db.Exec(query)
	if err != nil {
		log.Fatal(err)
		return "NO"
	}
	return "OK"
}

func Update(db *sql.DB, table string, data map[string]string, where map[string]string) (status string) {
	set := ""
	condition := ""
	for f, v := range data {
		set += fmt.Sprintf(", %s='%s'", f, v)
	}
	for f, v := range where {
		condition += fmt.Sprintf(" AND %s='%s'", f, v)
	}
	query := fmt.Sprintf("UPDATE %s SET %s WHERE %s", table, set[1:], condition[4:])
	_, err := db.Exec(query)
	if err != nil {
		log.Fatal(err)
		return "NO"
	}
	return "OK"
}

func Delete(db *sql.DB, table string, where map[string]string) (status string) {
	query := "DELETE FROM " + table
	if where != nil {
		condition := ""
		for f, v := range where {
			condition += fmt.Sprintf(" AND %s='%s'", f, v)
		}
		query += " WHERE " + condition[4:]
	}
	_, err := db.Exec(query)
	if err != nil {
		log.Fatal(err)
		return "NO"
	}
	return "OK"
}

func Fetch(db *sql.DB, table string, where map[string]string) (result map[string]string, err error) {
	query := "SELECT * FROM " + table
	if where != nil {
		condition := ""
		for f, v := range where {
			condition += fmt.Sprintf(" AND %s='%s'", f, v)
		}
		query += " WHERE " + condition[4:]
	}
	rows, err := db.Query(query)
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	// Retrieve column names
	columns, err := rows.Columns()
	if err != nil {
		return nil, err
	}

	// Assume only one row is returned, otherwise handle it appropriately
	if rows.Next() {
		values := make([]interface{}, len(columns))
		for i := range values {
			values[i] = new(sql.RawBytes)
		}
		err = rows.Scan(values...)
		if err != nil {
			return nil, err
		}

		result = make(map[string]string)
		for i, colName := range columns {
			result[colName] = string(*values[i].(*sql.RawBytes))
		}
	} else {
		return nil, fmt.Errorf("no rows found")
	}

	return result, nil
}

func Select(db *sql.DB, table string, where map[string]string, order string, limit int) (results []map[string]string) {
	query := "SELECT * FROM " + table
	if where != nil {
		condition := ""
		for f, v := range where {
			condition += fmt.Sprintf(" AND %s='%s'", f, v)
		}
		query += " WHERE " + condition[4:]
	}
	if order != "" {
		query += " ORDER BY " + order
	}
	if limit != 0 {
		query += fmt.Sprintf(" LIMIT %d", limit)
	}
	rows, err := db.Query(query)
	if err != nil {
		log.Fatal(err)
	}
	defer rows.Close()
	columns, err := rows.Columns()
	if err != nil {
		log.Fatal(err)
	}
	values := make([]interface{}, len(columns))
	for i := range values {
		values[i] = new(sql.RawBytes)
	}
	for rows.Next() {
		err = rows.Scan(values...)
		if err != nil {
			log.Fatal(err)
		}
		result := make(map[string]string)
		for i, colName := range columns {
			result[colName] = string(*values[i].(*sql.RawBytes))
		}
		results = append(results, result)
	}
	if err = rows.Err(); err != nil {
		log.Fatal(err)
	}
	return results
}

func Rowcount(db *sql.DB, table string, where map[string]string) (count int) {
	query := "SELECT COUNT(*) FROM " + table
	if where != nil {
		condition := ""
		for f, v := range where {
			condition += fmt.Sprintf(" AND %s='%s'", f, v)
		}
		query += " WHERE " + condition[4:]
	}
	err := db.QueryRow(query).Scan(&count)
	if err != nil {
		log.Fatal(err)
	}
	return count
}

func Truncate(db *sql.DB, table string) (status string) {
	query := "TRUNCATE " + table
	_, err := db.Exec(query)
	if err != nil {
		log.Fatal(err)
		return "NO"
	}
	return "OK"
}

func main() {

}

File: config/function_crud.php
<?php    
    function insert($conn, $table, $data=null) {
        $command = 'INSERT INTO '.$table;
        $field = $value = null;
        foreach($data as $f => $v) {
            $field .= ','.$f;
            $value .= ", '".$v."'";
        }
        $command .= ' ('.substr($field,1).')';
        $command .= ' VALUES('.substr($value,1).')';
        $exec = pg_query($conn, $command);
        ($exec) ? $status = 'OK' : $status = 'NO';
        return $status;
    }
    
    function update($conn, $table, $data=null, $where=null) {
        $command = 'UPDATE '.$table.' SET ';
        $field = $value = null;
        foreach($data as $f => $v) {
            $field .= ",".$f."='".$v."'";
        }
        $command .= substr($field,1);
        if($where!=null) {
            foreach($where as $f => $v) {
                $value .= "#".$f."='".$v."'";
            }
            $command .= ' WHERE '.substr($value,1);
            $command = str_replace('#',' AND ',$command);
        }
        $exec = pg_query($conn, $command);
        ($exec) ? $status = 'OK' : $status = 'NO';
        return $status;
    }
    
    function delete($conn, $table, $where=null) {
        $command = 'DELETE FROM '.$table;
        if($where!=null) {
            $value = null;
            foreach($where as $f => $v) {
                $value .= "#".$f."='".$v."'";
            }
            $command .= ' WHERE '.substr($value,1);
            $command = str_replace('#',' AND ',$command);
        }
        $exec = pg_query($conn, $command);
        ($exec) ? $status = 'OK' : $status = 'NO';
        return $status;
    }
    
    function fetch($conn, $table, $where=null) {
        $command = 'SELECT * FROM '.$table;
        if($where!=null) {
            $value = null;
            foreach($where as $f => $v) {
                $value .= "#".$f."='".$v."'";
            }
            $command .= ' WHERE '.substr($value,1);
            $command = str_replace('#',' AND ',$command);
        }
        $sql = pg_query($conn, $command);
        $exec = pg_fetch_assoc($sql);
        return $exec;
    }
    
    function select($conn, $table, $where=null, $order=null, $limit=null) {
        $command = 'SELECT * FROM '.$table;
        if($where!=null) {
            $value = null;
            foreach($where as $f => $v) {
                $value .= "#".$f."='".$v."'";
            }
            $command .= ' WHERE '.substr($value,1);
            $command = str_replace('#',' AND ',$command);
        }
        ($order!=null) ? $command .= ' ORDER BY '.$order :null;
        ($limit!=null) ? $command .= ' LIMIT '.$limit :null;
        $result = array();
        $sql = pg_query($conn, $command);
        while($field = pg_fetch_assoc($sql)) {
            $result[] = $field;
        }
        return $result;
    }
    
    function rowcount($conn, $table, $where=null) {
        $command = 'SELECT COUNT(*) FROM '.$table;
        if($where!=null) {
            $value = null;
            foreach($where as $f => $v) {
                $value .= "#".$f."='".$v."'";
            }
            $command .= ' WHERE '.substr($value,1);
            $command = str_replace('#',' AND ',$command);
        }
        $exec = pg_query($conn, $command);
        $row = pg_fetch_row($exec);
        return $row[0];
    }
    
    function truncate($conn, $table) {
        $command = 'TRUNCATE '.$table;
        $exec = pg_query($conn, $command);
        ($exec) ? $status = 'OK' : $status = 'NO';
        return $status;
    }
?>
